esmvaltool/diag_scripts/lifetime/lifetime.py

```diff
@@ -0,0 +1,1498 @@
+                iris.plot.plot(annual_mean_cube, **plot_kwargs)
+            else:
+                raise ValueError(f"uknown option for annual_mean."
+                                 "Choose False, 'both', or 'only'.")
+
+        # Default plot appearance
+        multi_dataset_facets = self._get_multi_dataset_facets(
+            list(base_datasets.values()))
+        axes.set_title(f'{self.names["long_name"]} in region {region}')
+        axes.set_xlabel('Time')
+        axes.set_ylabel(f"{chr(964)}({self._get_name('reactant').upper()})"
+                        f" [{self.units}]")
+        gridline_kwargs = self._get_gridline_kwargs(plot_type)
+        if gridline_kwargs is not False:
+            axes.grid(**gridline_kwargs)
+
+        # Legend
+        legend_kwargs = self.plots[plot_type]['legend_kwargs']
+        if legend_kwargs is not False:
+            axes.legend(**legend_kwargs)
+
+        # Customize plot appearance
+        self._process_pyplot_kwargs(plot_type, multi_dataset_facets)
+
+        # Save plot
+        plot_path = self.get_plot_path(plot_type, multi_dataset_facets)
+        fig.savefig(plot_path, **self.cfg['savefig_kwargs'])
+        logger.info("Wrote %s", plot_path)
+        plt.close()
+
+        # Save netCDF file
+        netcdf_path = get_diagnostic_filename(Path(plot_path).stem, self.cfg)
+        var_attrs = {
+            'short_name': self.names['short_name'],
+            'long_name': self.names['long_name'],
+            'units': self.units
+        }
+        io.save_1d_data(cubes, netcdf_path, 'time', var_attrs)
+
+        # Provenance tracking
+        caption = (f"Time series of {multi_dataset_facets['long_name']} for "
+                   f"various datasets.")
+        provenance_record = {
+            'ancestors': ancestors,
+            'authors': ['schlund_manuel', 'winterstein_franziska'],
+            'caption': caption,
+            'plot_types': ['line'],
+            'long_names': [var_attrs['long_name']],
+        }
+        with ProvenanceLogger(self.cfg) as provenance_logger:
+            provenance_logger.log(plot_path, provenance_record)
+            provenance_logger.log(netcdf_path, provenance_record)
+
+    def create_annual_cycle_plot(self, region, input_data, base_datasets):
+        """Create annual cycle plot."""
+        plot_type = 'annual_cycle'
+        if plot_type not in self.plots:
+            return
+
+        if not base_datasets:
+            raise ValueError(f"No input data to plot '{plot_type}' given")
+
+        logger.info("Plotting %s", plot_type)
+        fig = plt.figure(**self.cfg['figure_kwargs'])
+        axes = fig.add_subplot()
+
+        # Plot all datasets in one single figure
+        ancestors = []
+        cubes = {}
+        for label, dataset in input_data.items():
+            _ = [ancestors.append(variable['filename'])
+                 for variable in dataset['dataset_data']]
```

Comment:
```suggestion
            ancestors.extend(variable['filename'] for variable in dataset['dataset_data'])
```

Also applies to other similar lines of code